### PR TITLE
Add internal consistency of descriptions concept

### PR DESCRIPTION
**Pull request by cornelius**
This change adds the document describing the concept how we deal
with the internal consistency of system descriptions. It is the
result of a series of discussions in the team.

The document outlines the current state, the use cases, and the
future direction. It is meant as a base for future implementation
work, and also contains the plan for some gradual changes to the
structure of the scopes.

This pull request obsoletes #725.

### CODE REVIEW

File: docs/Internal-Consistency-of-System-Descriptions.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+To make it explicit what dependencies and implications a scope has we add an explicit definition of a scope. This is a YAML file, which defines, what filters to apply, what belongs to the scope and other meta data.
+
+#### Filter definition
+
+An example for a minimal filter definition based on the current functionality is a file `scopes/repositories.yam` with the content:
+
+```yaml
+filters:
+  export:
```
Comment (aduffeck) [inline docs/Internal-Consistency-of-System-Descriptions.md]: Based on our discussion yesterday, do we want to explicitly disallow `import` filters?

Comment (aduffeck) [inline docs/Internal-Consistency-of-System-Descriptions.md]: `inspect` filters, sorry...

Comment (cornelius) [inline docs/Internal-Consistency-of-System-Descriptions.md]: I will add a note about that.

**Comment by aduffeck** [comment]
lgtm now

**Pull request merged**